### PR TITLE
Daily Double

### PR DESCRIPTION
**Pull request by ryan-novak**
## What does this PR do?
This PR completes the functionality of the daily double turns, and also fixes an issue where there was endless daily double turns happening.

## Why are we doing this? Any context or related work?
To complete the user story around the daily double functionality.

## Where should a reviewer start?
Changes were made in the index.js file.

https://github.com/VeeAndrade/Game-time/projects/1

@hannahhch
@khalidwilliams

Hey fearless leaders, I think we are half a day late on tagging both of you in this PR.... I would say we have about 85%-90% of the game complete, just some minor stuff to clean up and then a decent amount of refactoring is our remaining to-do's. Any feedback would be greatly appreciated!

### CODE REVIEW

File: src/Daily-Double.js [hunk 1]
```diff
@@ -8,14 +8,18 @@ class DailyDouble extends Clue{
...
+      console.log(this.ddPointValue)
     }
-    if (player.score > cluePoints) {
+    if (player.score > this.ddPointValue) {
       wagerLimit = player.score
     } else {
-      wagerLimit = cluePoints;
+      wagerLimit = this.ddPointValue;
     }
+    console.log('skldaf;saf', wagerLimit)
```
Comment (khalidwilliams) [inline src/Daily-Double.js]: Make sure to take out console logs before submission! It's generally a good practice to not leave them in mergeable changes

File: src/Daily-Double.js [hunk 2]
```diff
@@ -8,14 +8,18 @@ class DailyDouble extends Clue{
 
   determineWager(turns, player, cluePoints) {
     let wagerLimit = 0;
+    this.ddPointValue = cluePoints;
+    // debugger
```
Comment (khalidwilliams) [inline src/Daily-Double.js]: Same thing, remove commented out code before merging

File: src/Daily-Double.js [hunk 3]
```diff
@@ -8,14 +8,18 @@ class DailyDouble extends Clue{
...
-      this.pointValue = this.pointValue * 2;
+      this.ddPointValue = cluePoints * 2;
+      console.log(this.ddPointValue)
     }
-    if (player.score > cluePoints) {
+    if (player.score > this.ddPointValue) {
       wagerLimit = player.score
     } else {
-      wagerLimit = cluePoints;
+      wagerLimit = this.ddPointValue;
```
Comment (khalidwilliams) [inline src/Daily-Double.js]: Will the `ddPointValue` always be the highest remaining score on the board? If not, you may need to pass the highest remaining value on the board into this method on line 9

File: src/index.js [hunk 4]
```diff
@@ -258,6 +262,10 @@ function makeDailyDouble(player) {
   // console.log(dailyDouble)
   let highestPointClue = sortClues();
   wagerAmount = dailyDouble.determineWager(turns, player, highestPointClue);
+  console.log(dailyDouble)
+  console.log(turns)
+  console.log(highestPointClue)
+  console.log(wagerAmount)
```
Comment (khalidwilliams) [inline src/index.js]: Be careful not to leave these in during submission (assuming they aren't needed for the purposes of your demo)

File: src/Daily-Double.js (hunk 3 above)
Comment (khalidwilliams) [reply src/Daily-Double.js]: Just kidding, I see how you're passing in the highest clue value in the index.js file. This parameter name `cluePoints` is a little unclear, but I see what you're doing now.

File: src/index.js [hunk 5]
```diff
@@ -53,7 +54,9 @@ let submitWagerBtn = document.querySelector(".submit-wager");
 let submitFinalBtn = document.querySelector(".submit-final");
 let totalClues;
 let wagerAmount;
-let ddGuessBtn = document.querySelector('.submit-DD-wager')
+let playersWager;
+let ddWagerBtn = document.querySelector('.submit-DD-wager');
+let ddSubmitBtn = document.querySelector('.daily-double-guess-btn');
```
Comment (khalidwilliams) [inline src/index.js]: You can use jQuery for your query selecting too!

File: src/index.js [hunk 6]
```diff
@@ -235,7 +239,7 @@ function displaySelectedClue(event) {
   selectedClue = clueInfo.find(clue => clue.id == clickedCard.id)
   clueCards.classList.add('no-clicks');
   turns ++;
-  if (turns === randomNumber1 || randomNumber2 || randomNumber3) {
+  if (turns === randomNumber1 || turns === randomNumber2 || turns === randomNumber3) {
```
Comment (khalidwilliams) [inline src/index.js]: This is interesting -- are you creating the daily doubles based on turn number and not clue position?

File: src/index.js [hunk 7]
```diff
@@ -268,19 +276,41 @@ function sortClues() {
...
+  $('.selected-clue-info').css("display", "none");
+  $('.game-categories').css("display", "none");
 }
 
-function createDailyDouble() {
-  $('.daily-double-wager-amount').text(`Set your wager between 5 and ${wagerAmount} points.`)
+function evaluateDailyDoubleGuess() {
+  if ($('.daily-double-input').val() === '') {
+    return $('.guess-error').text("Please enter your guess below")
+  } if ($('.daily-double-input').val().toUpperCase() === clue.answer.toUpperCase()) {
```
Comment (khalidwilliams) [inline src/index.js]: unless this is meant to be an `else if`, I'd put the `if` statement on a new line. It just feels more readable

File: src/Daily-Double.js (hunk 3 above)
Comment (ryan-novak) [reply src/Daily-Double.js]: Yes I agree, variable names are very iffy and will be changed to be more logical.

File: src/index.js (hunk 6 above)
Comment (ryan-novak) [reply src/index.js]: Yes, it's based on the turns. When we began this functionality, we never made an array of all total clues, so at the start of each turn we increment the turn count and if that matches a random number, it ends up being a daily double.

File: src/index.js (hunk 7 above)
Comment (ryan-novak) [reply src/index.js]: Yes, it should be a new line. Thanks for pointing that out.

**Pull request merged**